tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -8535,6 +8535,102 @@ public function testDynamicConstantsWithNativeTypes(
 		);
 	}
 
+	public function dataMixedTypeConstants(): array
+	{
+		return [
+			[
+				'mixed',
+				'MixedTypeConstants\MixedTypeConstantClass::MIXED_TYPE_CONSTANT_IN_CLASS',
+			],
+			[
+				"'abc123def'",
+				'MixedTypeConstants\MixedTypeConstantClass::PURE_CONSTANT_IN_CLASS',
+			],
+			[
+				"'xyz'",
+				'MixedTypeConstants\NoMixedTypeConstantClass::MIXED_TYPE_CONSTANT_IN_CLASS',
+			],
+			[
+				'false',
```

Comment:
This doesn't seem right to me, but this test is copied and modified from `testDynamicConstants` where it also has `false`. That changed from `bool` to `false` in https://github.com/phpstan/phpstan-src/commit/285b49f2072e18ffe9ff72f221151512b8a92fe6#diff-52d2cce81a3daff27b7d6d8ba619b045420ab9a9acffc57c1c379af4eba0b127R8384.

I tried tracking down how I might be able to fix it. I found that the constant gets defined here:
https://github.com/phpstan/phpstan-src/blob/d73acef87f9d3ebd04bb6671d6b83755a8e62e8e/src/Type/Php/DefineConstantTypeSpecifyingExtension.php#L45-L60

The issue is that `$scope->getType($node->getArgs()[1]->value)` is just going based off the value, so it never reaches where I think it would determine that the constant is dynamic:
https://github.com/phpstan/phpstan-src/blob/d73acef87f9d3ebd04bb6671d6b83755a8e62e8e/src/Analyser/MutatingScope.php#L1943